### DISCUSSION TITLE
Informal "chat" thread about current coupled runs

### DISCUSSION THREAD

**Discussion by PeterHjortLauritzen** [NCAR/amwg_dev#455] [Discussion about coupled runs]
This is a continuous discussion thread about current coupled runs (replacing some Google chat threads)

**Comment by cecilehannay** [comment]
Based on today discussion, we are starting 2 branches:  Cold branch / Warm branch 

|                  | Cold branch    | Warm branch    |
| -------------    | -------------  | -------------  |
| clubb diffusion  | on             | off ( default) |
| tau code         | off (default)  | on             | 
| gustiness        | on             | on             | 
| NorESM mods      | off            | off            | 
| New PUMAS limiter| on             | on             | 
| Tuning           | use 68 tuning  | Ben's tuning   | 

We will also continue the 64 perturbations 
- 64 => froze over
- 64e, 64f start a branch from 64 at year 43 => froze over
- 64g, 64h start a branch from 64 at year 33

**Comment by PeterHjortLauritzen** [comment]
From @adamrher :  "It looks like the warm branch is too warm and the cold branch is way too cold (it already froze)." -> we need to retune ...

**Reply by adamrher** [reply]
orange is cold branch, red the warm branch.

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/1b57a560-c850-4d70-8c78-5913ad5d8f9e)

**Reply by adamrher** [reply]
SWCF and LWCF included:

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/2d02de13-1674-4adf-8459-39911924bb2e)

**Reply by adamrher** [reply]
SWCF maps (first 20 years)

![temp_SWCF](https://github.com/NCAR/amwg_dev/assets/29961476/f4369307-eda0-4e91-839b-7beded066b5f)

SST bias relative to HadISST PI (first 20 years)

![temp_SST](https://github.com/NCAR/amwg_dev/assets/29961476/76e4bcc8-0d48-48f9-9211-44d1b4e30018)

**Comment by swrneale** [comment]
A few questions/discussion:
1. Regionally, where are the SSTs cooling/warming the most in 73/74?
2. What aspects are mostly causing the cold/warm contrasts (diffusion/tau/tuning?).
3. Why are we starting the runs from the same time each time? Seriously? Through year 50 the reasonable runs have the same seemingly inevitable SST RESTOM inconsistency (RESTOM about right , global SST decreasing). After that 54 reaches a more reasonable equilibrium between the two 26g is even more extreme. The configuration pumps 0.5W of heat into the ocean for 70 years and still it cools.

**Comment by cecilehannay** [comment]
Updates:

Tuning of cold/warm branch:
- I started a retuned version of the cold branch: https://github.com/NCAR/amwg_dev/issues/462 (we got 6 years so far)
- I can start a retuned version of the warm branch as soon as I get tuning knobs for that (@bstephens82). 

Following the CESM project meeting:
- We are extending 54 to 150 years as the end of 54 displayed possible hints of the Lb Sea starting to freeze.
- We started run 064intel (#464) to compare standard compiler (intel) and the 'new' intel one (intel-oneapi)

**Reply by adamrher** [reply]
Mini-update. 75 cold leg run looking ok. RESTOM averages to zero. 12 years done. Note the lime green line (75) is the only one w/o 5-year running avg; it's the year-to-year value of RESTOM.

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/79dadecc-5618-4c94-a756-73d4b8479d4c)

**Reply by adamrher** [reply]
run 75 is out to year 22, but I think it stopped. @cecilehannay can we make sure to resume that run?

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/7a1103bb-fd90-42d5-bc48-d59033f86726)

**Reply by cecilehannay** [reply]
Hi @adamrher: I just resubmitted 75. 

As a rule of the thumb: By default, I stopped the run at 22 years (except when we specify to make it shorter) so we can look at the first 20 years and decide if the run is ok and we want to make it longer.

**Comment by cecilehannay** [comment]
@adamrher: could you run your spaghetti plot for 64, 64e,f,g,h and 64intel?  
Thanks

**Reply by adamrher** [reply]
Here's a current look. The only branch of 64 that froze over is 64e. Let's carry on with these experiments.

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/b9cc59ab-6893-41f3-a207-649fccf27548)

**Reply by adamrher** [reply]
I was looking at these in bit more detail this morning. I even added the relevant part of 26g where it freezes over as well:

![temp_tseries_SST 64](https://github.com/NCAR/amwg_dev/assets/29961476/bf1a1c97-febb-4aa9-aee0-de3b3fed27a3)

(note I've changed up the colors). I think this plot emphasizes what you see in the annual plots, the global SST's are trending down for most of the simulations, but then around year 60-70 they start to reverse and trend up again. It looks like if a run can avoid freezing over during the coldest parts, year 40 to 70, then it's out of the wood work, at least until the next inter-decadal SST cooling.

It is interesting how 64f (yellow) barely avoids freezing over (see LabSea plots), whereas 64e (purple) couldn't make it. 64g and 64h, having been initialized ten years earlier, cruise right by without much of a threat of freezing.

**Comment by adamrher** [comment]
Here's the current status of the cold (75) and warm branch (76):

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/9d21088b-b07f-41da-aa6d-7b8cf2c8099c)

**Comment by adamrher** [comment]
Here's the latest spaghetti plot with Ben's run overlain. I am referring to it as run 77. I like how it doesn't have that big RESTOM bump in the beginning like we see in 76.

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/3d2a31d2-6b25-465c-9fd8-e3e4eff632d1)

@cecilehannay should we open an issue for Ben's run so it's documented officially as run 77?

**Reply by cecilehannay** [reply]
We have an issue for **77** alias **b1850.076base.n2th1** at #484.

**Comment by adamrher** [comment]
To take a closer look at the big RESTOM incursion around year 50 in run 77, here's a spaghetti plot w/o the 5 year running average for RESTOM.  It doesn't look so abnormal on a year by year basis.

![temp_tseries_SST clds](https://github.com/NCAR/amwg_dev/assets/29961476/66f4cbac-1b03-4522-a0b3-c22cc27843ba)

Here's the 5 year avg RESTOM plot, as a reminder:

![temp_tseries_SST](https://github.com/NCAR/amwg_dev/assets/29961476/54721849-4499-44b7-ad2f-17c80094d434)

**Comment by cecilehannay** [comment]
Urgent: I filled my quota (200T) and I needed to do a major clean up before I can run anything again. I can move  some case to `campaign` but not **everything**. Otherwise, we will fill campaign in no time. 
Please indicate what you want to keep asap. (I will do my best guess and start cleaning today). 

```
b.e23_alpha16b.BLT1850.ne30_t232.033
b.e23_alpha16b.BLT1850.ne30_t232.034
b.e23_alpha16b.BLT1850.ne30_t232.035
b.e23_alpha16b.BLT1850.ne30_t232.036
b.e23_alpha16b.BLT1850.ne30_t232.037
b.e23_alpha16b.BLT1850.ne30_t232.038
b.e23_alpha16b.BLT1850.ne30_t232.039
b.e23_alpha16b.BLT1850.ne30_t232.040
b.e23_alpha16b.BLT1850.ne30_t232.041
b.e23_alpha16b.BLT1850.ne30_t232.042
b.e23_alpha16b.BLT1850.ne30_t232.043
b.e23_alpha16b.BLT1850.ne30_t232.044
b.e23_alpha16b.BLT1850.ne30_t232.048
b.e23_alpha16b.BLT1850.ne30_t232.051
b.e23_alpha16b.BLT1850.ne30_t232.052
b.e23_alpha16b.BLT1850.ne30_t232.053
b.e23_alpha16b.BLT1850.ne30_t232.054
b.e23_alpha16g.BLT1850.ne30_t232.058
b.e23_alpha16g.BLT1850.ne30_t232.060
b.e23_alpha16g.BLT1850.ne30_t232.061
b.e23_alpha16g.BLT1850.ne30_t232.062
b.e23_alpha16g.BLT1850.ne30_t232.063
b.e23_alpha16g.BLT1850.ne30_t232.064
b.e23_alpha16g.BLT1850.ne30_t232.064b
b.e23_alpha16g.BLT1850.ne30_t232.064c
b.e23_alpha16g.BLT1850.ne30_t232.064d
b.e23_alpha16g.BLT1850.ne30_t232.064e
b.e23_alpha16g.BLT1850.ne30_t232.064f
b.e23_alpha16g.BLT1850.ne30_t232.064g
b.e23_alpha16g.BLT1850.ne30_t232.064h
b.e23_alpha16g.BLT1850.ne30_t232.064i
b.e23_alpha16g.BLT1850.ne30_t232.064intel
b.e23_alpha16g.BLT1850.ne30_t232.064j
b.e23_alpha16g.BLT1850.ne30_t232.065
b.e23_alpha16g.BLT1850.ne30_t232.066
b.e23_alpha16g.BLT1850.ne30_t232.067
b.e23_alpha16g.BLT1850.ne30_t232.068
b.e23_alpha16g.BLT1850.ne30_t232.069
b.e23_alpha16g.BLT1850.ne30_t232.070
b.e23_alpha16g.BLT1850.ne30_t232.072
b.e23_alpha16g.BLT1850.ne30_t232.073
b.e23_alpha16g.BLT1850.ne30_t232.074
b.e23_alpha16g.BLT1850.ne30_t232.075
b.e23_alpha16g.BLT1850.ne30_t232.076
b.e23_alpha16g.BLT1850.ne30_t232.078
b.e23_alpha16g.BLT1850.ne30_t232.079
b.e23_alpha16g.BLT1850.ne30_t232.080
b.e23_alpha16g.BLT1850.ne30_t232.4CO2.078
```
For Fcase,
```
f.cam6_3_117.FLTHIST_ne30.r250.slowice.001
f.cam6_3_117.FLTHIST_ne30.r250.slowsmice.001
f.cam6_3_117.FLTHIST_ne30.r250.slowsnow.001
f.cam6_3_117.FLTHIST_ne30.r250.wsub_min0.002.001
f.cam6_3_119.FLTHIST_ne30.r328.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae_dst.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae_dst1.3.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae_numcin2.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae_numcin3.001
f.cam6_3_119.FLTHIST_ne30.r328_gamma0.33_soae_sfcflx.001
f.cam6_3_119.FLTHIST_ne30.r328.opt.001
f.cam6_3_119.FLTHIST_ne30.r328.opt.002
f.cam6_3_119.FLTHIST_ne30.r328-wsub_scale0.1.001
f.cam6_3_119.FLTHIST_ne30.r328-wsub_scale0.5.001
f.cam6_3_119.FLTHIST_ne30.r328-wsub_scale0.5.002
f.cam6_3_119.FMTHIST_ne30.r328.001
f.cam6_3_119.FMTHIST_ne30.r328_gamma0.33_soae.001
f.cam6_3_132.FLT2000.ne30.001
f.cam6_3_132.FLT2000.ne30.001_pia
f.cam6_3_132.FMTHIST_ne30.001
f.cam6_3_132.FMTHIST_ne30.001b
f.e23_alpha16b.FLT1850.ne30.033
f.e23_alpha16b.FLT2000.ne30.033
f.e23_alpha16b.FLT2000.ne30.033_pia
```

@PeterHjortLauritzen, @JulioTBacmeister,  @swrneale, @adamrher, @bstephens82, @gustavo-marques, @cmcclus @dlawrenncar @megandevlan @islasimpson

**Reply by cmcclus** [reply]
can I make a request to keep 
f.cam6_3_132.FLT2000.ne30.001
f.cam6_3_132.FLT2000.ne30.001_pia
and 
f.e23_alpha16b.FLT2000.ne30.033
f.e23_alpha16b.FLT2000.ne30.033_pia

let me know if these are too large  to keep...

**Reply by megandevlan** [reply]
I'd request to keep at least
b.e23_alpha16b.BLT1850.ne30_t232.054
b.e23_alpha16g.BLT1850.ne30_t232.064 

That's at least where my initial focus has been; the full 64 series would probably be good to keep too though, space permitting.

**Comment by bstephens82** [comment]
Sorry to add something that's not on your list, but I am wondering if we could keep my "077" run (which has an alias), since that's the only long taus run we have?  If so, the history files are here: /glade/derecho/scratch/stepheba/archive/b1850.076base.n2th1. If you do move it over to campaign storage, I assume you could rename the folder to the "official" 077 name.  I looked at the 080 run and it looks like it has stopped at 22 years...are we going to continue that one?  I don't know at this point if it's worth the resources or not considering the taus' disappointing performance at the Friday CAM7 meeting.  It looks like we have not found the right area of parameter space yet, to me.

**Reply by cecilehannay** [reply]
At the CAM7 meeting, Peter said it was up to you to decide if we want to continue 80 or not. I agree that it might be not worth the resources. 
Once we agree on a location where to put the coupled runs, you could move 77 into it. 
I am thinking to move the coupled runs into 
```
/glade/campaign/cesm/development/cross-wg/cesm3_development
```
as these a cross-working group activities.

But if you have not reached your quota, you should not be worried about moving it on campaign today. The issue are the ones that are in **my** archive.

**Reply by bstephens82** [reply]
Ok I will make a note to check back in a week or two and possibly move it over then, thanks.  I may comment on 80 on the other thread dedicated to that run.

**Comment by swrneale** [comment]
Is the data from all these simulations roughly the same, and which of the h1/h2/h3 file sizes are dominating?

**Reply by cecilehannay** [reply]
Roughly,
- Ocean is 67%
- Atmosphere is 32%

For atmosphere output:
- monthly (h0) -> 45%
- daily (h2) -> 30%
- 3-hourly (h3) -> 7%
- monthly (h4 -> TEM) -> small
- monthly (h6 -> aerosol) -> 18%

Obviously, it would help to reduce ocean output but I don't have much control on that. 
For the atmosphere, I am wondering if @tlimes really needs all these aerosol species in every run (we coudl save 18% of the atmosphere)

```
fincl1 = 'ACTNI', 'ACTNL', 'ACTREI', 'ACTREL', 'AODDUST', 'AODVIS', 'AODVISdn','BURDENBC', 'BURDENDUST', 'BURDENPOM', 'BURDENSEASALT', 
'BURDENSO4', 'BURDENSOA', 'CAPE', 'CCN3', 'CDNUMC', 'CH4', 'CLDHGH', 'CLDICE', 'CLDLIQ', 'CLDLOW', 'CLDMED', 'CLDTOT', 'CLOUD', 'CMFMC_DP', 
'CT_H2O', 'DCQ', 'DQCORE', 'DTCOND', 'DTCORE', 'DTV', 'EVAPPREC', 'EVAPSNOW', 'FCTI', 'FCTL', 'FICE', 'FLDS', 'FLNS', 'FLNSC', 'FLNT', 'FLNTC', 'FLUT', 
'FREQZM', 'FSDS', 'FSDSC', 'FSNS', 'FSNSC', 'FSNT', 'FSNTC', 'FSNTOA', 'ICEFRAC', 'LANDFRAC', 'LHFLX', 'LWCF', 'MPDICE', 'MPDLIQ', 'MPDQ', 'MPDT', 
'OCNFRAC', 'OMEGA', 'OMEGA500', 'PBLH', 'PHIS', 'PINT', 'PMID', 'PRECC', 'PRECL', 'PRECSC', 'PRECSL', 'PRECT', 'PS', 'PSL', 'PTEQ', 'PTTEND', 'Q', 
'QFLX', 'QRL', 'QRS', 'QTGW', 'RCMTEND_CLUBB', 'RELHUM', 'RVMTEND_CLUBB', 'SHFLX', 'SOLIN', 'SST', 'STEND_CLUBB', 'SWCF', 
'T', 'TAUX', 'TAUY', 'TFIX', 'TGCLDIWP', 'TGCLDLWP', 'TMQ', 'TREFHT', 'TS', 'TTGW', 'U', 'U10', 'UBOT', 'UTGWORO', 'UTGW_TOTAL', 
'V', 'VBOT', 'VTGWORO', 'VTGW_TOTAL', 'WPRTP_CLUBB', 'WPTHLP_CLUBB', 'Z3', 'ZMDQ', 'ZMDT', 'N2O', 'CO2','CFC11','CFC12', 
'AODVISdn','CCN3', 'CDNUMC', 'H2O', 'NUMICE', 'NUMLIQ','OMEGA500'

fincl3 = 'PRECT', 'PRECC', 'FLUT', 'U850', 'U200', 'V850', 'V200', 'OMEGA', 'PSL'

fincl4 =  'PRECC','PRECL'

fincl5 = 'Uzm','Vzm','Wzm','THzm', 'VTHzm','WTHzm','UVzm','UWzm'
phys_grid_ctem_nfreq=-6
phys_grid_ctem_zm_nbas=120
phys_grid_ctem_za_nlat=90

fincl7= 'AQSO4_H2O2','AQSO4_O3', 'bc_a1', 'bc_a4', 'dst_a1', 'dst_a2', 'dst_a3', 'ncl_a1',
'ncl_a1', 'ncl_a2', 'ncl_a3', 'pom_a1', 'pom_a4', 'so4_a1', 'so4_a2', 'so4_a3',
'soa_a1', 'num_a1', 'num_a2', 'num_a3', 'num_a4',
'bc_a1SFWET', 'bc_a4SFWET', 'dst_a1SFWET', 'dst_a2SFWET', 'dst_a3SFWET', 'ncl_a1SFWET',
'ncl_a2SFWET', 'ncl_a3SFWET', 'pom_a1SFWET', 'pom_a4SFWET', 'so4_a1SFWET', 'so4_a2SFWET', 'so4_a3SFWET', 'soa_a1SFWET',
'soa_a2SFWET', 'bc_c1SFWET', 'bc_c4SFWET', 'dst_c1SFWET', 'dst_c2SFWET', 'dst_c3SFWET', 'ncl_c1SFWET', 'ncl_c2SFWET',
'ncl_c3SFWET', 'pom_c1SFWET', 'pom_c4SFWET', 'so4_c1SFWET', 'so4_c2SFWET', 'so4_c3SFWET', 'soa_c1SFWET', 'soa_c2SFWET',
'bc_a1DDF', 'bc_a4DDF', 'dst_a1DDF', 'dst_a2DDF', 'dst_a3DDF', 'ncl_a1DDF', 'ncl_a2DDF', 'ncl_a3DDF',
'pom_a1DDF', 'pom_a4DDF', 'so4_a1DDF', 'so4_a2DDF', 'so4_a3DDF', 'soa_a1DDF', 'soa_a2DDF',
'so4_a1_CLXF', 'so4_a2_CLXF', 'SFbc_a4', 'SFpom_a4', 'SFso4_a1', 'SFso4_a2',
'so4_a1_sfgaex1', 'so4_a2_sfgaex1', 'so4_a3_sfgaex1', 'soa_a1_sfgaex1', 'soa_a2_sfgaex1',
'SFdst_a1','SFdst_a2', 'SFdst_a3', 'SFncl_a1', 'SFncl_a2', 'SFncl_a3',
'num_a2_sfnnuc1', 'SFSO2', 'OCN_FLUX_DMS', 'SAD_SULFC', 'SAD_TROP', 'SAD_AERO'
```

**Comment by swrneale** [comment]
The questions might be whether the oceanographers want much of their data - probably not right?

**Comment by adamrher** [comment]
I think we can delete every coupled simulation **prior** to 54. @gustavo-marques is currently storing run's 46, 47 (default vs hycom1 vertical coordinate) which I'd like to keep around since they are 100+ years long. He is also storing 45 and 55 (1/4deg MOM, and hycom1 version of 54). I can't think of a reason to keep any other pre-54 simulation around.

I'd prefer to keep the F cases around, at least until they're scrubbed by Derecho.

**Reply by gustavo-marques** [reply]
@adamrher: your suggestion sounds good. 
@cecilehannay: I can store 54 if that would be helpful.

**Reply by cecilehannay** [reply]
My plan is to save the runs we want to keep in the cross-working group space::
```
/glade/campaign/cesm/development/cross-wg/cesm3_development
```

**Reply by cecilehannay** [reply]
I will delete
```
b.e23_alpha16b.BLT1850.ne30_t232.033
b.e23_alpha16b.BLT1850.ne30_t232.034
b.e23_alpha16b.BLT1850.ne30_t232.035
b.e23_alpha16b.BLT1850.ne30_t232.036
b.e23_alpha16b.BLT1850.ne30_t232.037
b.e23_alpha16b.BLT1850.ne30_t232.038
b.e23_alpha16b.BLT1850.ne30_t232.039
```
It will allow me to restart the coupled simulation.

**Comment by swrneale** [comment]
Are you still keeping the climo files for all the runs we have done. I think these are very useful as quite often we end circling back around to revisit tuning/changes we have already attempted and it is good to have these files for guidance.

**Reply by cecilehannay** [reply]
Yes, I am keeping climos each time I ran the ADF in 
```
/glade/campaign/cgd/amp/amwg/climo
``` 
I believe @justin-richling is doing that too.

**Comment by gustavo-marques** [comment]
Frank wants to keep the 64 series to help us better understand the freezing of the Lab Sea. @cecilehannay, please let me know if you want me to store these cases in the OMWG storage space.

**Reply by cecilehannay** [reply]
@gustavo-marques: My thinking is that the best location for these runs is in the cross-working group space: 
```
/glade/campaign/cesm/development/cross-wg/cesm3_development
```
